### PR TITLE
Derelict porting

### PR DESCRIPTION
**Pull request by jbox144**
A port of the /TG/ style derelict to the AuStation codebase

## About The Pull Request

This PR overhauls "The Derelict" (Also known as KS13, or Kom... Kosmi... Kos... uhhh... "Kosmicheskaya Stantsiya 13") by porting it from the /TG/ codebase. This version of the map adds in lots of new features such as the drone vault, a power puzzle which requires the stations power network be repaired to open the internal vault and reap the valuable materials within.

The /TG/ variant of KS13 even includes a, limited, air tank for use in rebuilding the station. However, it is worth mentioning that this tank does not have an air pump within it, and this may just be a pipe dream (no pun intended).

It is worth mentioning that, currently, KS13 has no bluespace crystals on it, limiting them to asteroid mining. This also means drones will not be able to make bluespace miners or shuttles. If the drones run out of materials you effectively are back at square one. This may be changed in the near future once I start collating the drone changes I intend to work on.

I've also thrown in a couple of my own changes, partly because of the inconsistency of porting things. For example, the return of the KS13 plaque to mark the station. As well as another puzzle involving a broken down solar panel, where you use parts in an earlier segment to repair it. Nothing that challenging but it's interesting nonetheless.

## Why It's Good For The Game

At the moment, the KS13 spawn is *very very dull*. Once you finish redoing the solars, there's nothing really to do. You can theoretically scour space for materials, but the lack of any methods of space travel for drones restricts that into leaps of faith into the depths of space in hopes of colliding with something. The introduction of the vault provides incentive (and supplies) to stay around for a while and start producing the tools needed for space exploration.

Not only that, but a few extra still functioning facilities on the new KS13 mean drones can do things they otherwise couldn't before. It even has an air tank, meaning repressurizing the station is possible [citation needed], or perhaps cannibalizing the stations excessively reinforced hull for plasma to make tritium, maybe even fusion. Who knows? But the additional materials to start with make all of this easier in the long run.

## Changelog
:cl:
add: Readded the vault to KS13
add: Readded the KS13 plaque to KS13
add: Readded the air tank to KS13
tweak: Tweaked the KS13 map
code: Rebuilt tgui for the KS13 vault controller. If I fucked this up, let me know.
code: Editing the derelict has changed the formatting, and the /TG/ varient uses prefabs much more heavily, hence why there appear to be so many deletions. The map should still work.
admin: Derelict drones do not have access to diamonds or bluespace crystals. If they ask for crystals, consider spawning some in.
/:cl:

### CODE REVIEW

**Comment by MCterra10** [comment]
somebody didn't modularize their code

**Comment by totallynotkatfood** [comment]
big oof

**Comment by jbox144** [comment]
wait what

**Comment by thelaughingbomb** [comment]
You might be better off pushing it to upstream unless you want to go through and modularize everything here

**Comment by jbox144** [comment]
Can someone explain what this means?

**Comment by totallynotkatfood** [comment]
Create a pr on beestation hornet instead and the change will be passed on to here

**Comment by coolsurf6** [comment]
Pretty much austation doesn't touch beestation code, and only adds custom stuff to the austation folder itself so changes to beestation code doesn't caue conflicts. 

You would need to add all your changes and have them work in teh austation directories or surround your code stating it is custom austation code so resolving conflicts are easier. Probably best to go straight to bee and have them add it. If they don't ask some of the coders or codermins on modularizing.

**Comment by coolsurf6** [comment]
With a map change, modularizing might be more difficult. Don't quote me on that.

**Comment by jbox144** [comment]
_**Wish someone had told me sooner map changes are basically impossible because now I wasted about 10 hours**_

**Comment by jbox144** [comment]
Will it work if I move the code changes to AuStation's folder?

**Comment by MCterra10** [comment]
the map changes are fine since it's a ruin, not a station at all

**Comment by totallynotkatfood** [comment]
It’s just the spaceruins code and the markings code that needs to be modularized

**Comment by jbox144** [comment]
Ok, sounds easy enough

**Comment by Mick1299** [comment]
We keep all code changes in the AuStation file, take a look at any of the prs that merged for examples. This doesn't look like it would be too hard to modularise

**Comment by Mick1299** [comment]
Oh boy, should have refreshed before posting. Kat essentially hit the nail on the head, only the decals need to be modularised. Pm me of you need any help with it

**Comment by jbox144** [comment]
Pardon the two commits, I'm pretty terrible at using Github.

**Comment by MCterra10** [comment]
also conflict

**Comment by jbox144** [comment]
If I knew how to mapmerge I would.
And the conflict is in the tgui bundle, which I have no idea how to fix since the bundle is a compressed JS file.

**Comment by jbox144** [comment]
If anyone more skilled at byond/github/python/node.js wants to pick up on this for me, feel free

**Comment by jbox144** [comment]
Alright, after huddling in a corner for 2 days I manned up and did the thing

**Comment by TheFakeElon** [comment]
Honestly, you should PR this upatream, considering the 3000 line map changes and TGUI stuff

**Comment by jbox144** [comment]
I realise that now but I'm commited (No pun intended)

**Comment by jbox144** [comment]
Plus if any of you can tell me how to move everything to another codebase, that'd be great

**Comment by TheFakeElon** [comment]
But you're not the one who has to deal with conflicts of bee ever decide so to mess with derelict, couldn't you make a new austation derelict file, similarly to how we made a new austation (metastation) map

File: austation/code/game/objects/effects/decals/turfdecal/markings.dm [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+/obj/effect/turf_decal/derelict/twelve
+	icon_state = "derelict12"
+
+/obj/effect/turf_decal/derelict/thirteen
+	icon_state = "derelict13"
+
+/obj/effect/turf_decal/derelict/fourteen
+	icon_state = "derelict14"
+
+/obj/effect/turf_decal/derelict/fiveteen
```
Comment (Mick1299) [inline austation/code/game/objects/effects/decals/turfdecal/markings.dm:46]: Fifteen

**Comment by Mick1299** [comment]
You could do what Elon said, call this new derelict AuDerelict. Then replace the random space ruin code that references Derelict with AuDerelict and add a comment about the change

**Comment by jbox144** [comment]
I've already done enough for this as-is

**Comment by TheFakeElon** [comment]
PR to bee or do what Mick said, Both are _pretty_ easy.

**Comment by jbox144** [comment]
alright I think I'll take the option of moving this too bee. It was a good learning experience (that's sarcasm).

**Pull request closed without merging**